src/test/clojure/xtdb/tx_sink_kafka_test.clj

```diff
@@ -0,0 +1,117 @@
+(ns ^:kafka xtdb.tx-sink-kafka-test
+  (:require [clojure.test :as t]
+            [next.jdbc :as jdbc]
+            [xtdb.serde :as serde]
+            [xtdb.node :as xtn]
+            [xtdb.test-util :as tu]
+            [xtdb.util :as util]
+            [xtdb.tx-sink.main :as tx-sink])
+  (:import (org.apache.kafka.clients.consumer ConsumerRecord KafkaConsumer)
+           org.testcontainers.kafka.ConfluentKafkaContainer
+           org.testcontainers.utility.DockerImageName))
+
+(def ^:private ^:dynamic *bootstrap-servers* nil)
+
+(defonce ^ConfluentKafkaContainer kafka-container
+  (ConfluentKafkaContainer. (DockerImageName/parse "confluentinc/cp-kafka:7.8.0")))
+
+(t/use-fixtures :once
+  (fn [f]
+    (tu/with-container kafka-container
+      (fn [^ConfluentKafkaContainer c]
+        (binding [*bootstrap-servers* (.getBootstrapServers c)]
+          (f))))))
+
+(defn consume-messages [bootstrap-servers topic]
+  (let [props {"bootstrap.servers" bootstrap-servers
+               "group.id" (str "test-consumer-" (random-uuid))
+               "key.deserializer" "org.apache.kafka.common.serialization.ByteArrayDeserializer"
+               "value.deserializer" "org.apache.kafka.common.serialization.ByteArrayDeserializer"
+               "auto.offset.reset" "earliest"}
+        ^KafkaConsumer consumer (KafkaConsumer. ^java.util.Map props)
```

Comment:
I know it's 'only a test', but best to use `with-open` here, make it 'obviously no bugs' 🙂